src/runtime/d3d12compute.cpp

```diff
@@ -2711,6 +2778,16 @@ WEAK int halide_d3d12compute_device_release(void *user_context) {
     if (device) {
         d3d12compute_device_sync_internal(device, NULL);
 
+        for (int i = 0; i < MaxFrames; ++i) {
+            d3d12_frame *frame = &frame_pool[i];
+            release_object(frame);
```

Comment:
Does `release_object()` set the argument to null? If not, shouldn't we?

Reply:
In this case, no.
the frame pool is allocated statically when the runtime module is loaded:
```
WEAK d3d12_frame frame_pool[MaxFrames] = {};
```

Reply:
No, what I mean is that after the call to `release_object()`, the contents of `frame_pool[i]` are now invalid (the internal pointers refer to a released object). Shouldn't we explicitly clear `frame_pool[i]` to zero to ensure we don't accidentally reuse a stale object? eg

```
template<typename d3d12_T>
static void release_object(d3d12_T *obj) {
    TRACELOG;
    if (!obj) {
        TRACEPRINT("null object -- nothing to be released.\n");
        return;
    }
    release_d3d12_object(obj);
    memset(obj, 0, sizeof(*obj));  // <-- THIS
}
```

Reply:
Ah, you're right, I thought I was resetting the fields in the corresponding `release_d3d12_object<d3d12_frame>` function, but I was only resetting `frame->fence_signal`.